emily/handler/src/main.rs

```diff
@@ -1,34 +1,23 @@
-use crate::config::SETTINGS;
-use reqwest::Client;
+//! Emily API entrypoint.
+
+use api::handlers;
 use sbtc_common::logging::setup_logging;
-use std::net::ToSocketAddrs;
-use tracing::{error, info};
 use warp::Filter;
 
 mod api;
-mod client;
 mod common;
-mod config;
 
 #[tokio::main]
 async fn main() {
     setup_logging(false);
 
-    let client = Client::new();
-
-    let api_routes = api::routes::routes(client);
-    let routes = api_routes.with(warp::log("api"));
-
-    let addr_str = format!("{}:{}", SETTINGS.server.host, SETTINGS.server.port);
-    info!("Server will run on {}", addr_str);
+    let routes = api::routes::routes()
+        .recover(handlers::handle_rejection)
+        .with(warp::log("api"));
 
-    let addr = match addr_str.to_socket_addrs() {
-        Ok(mut addrs) => addrs.next().expect("No addresses found"),
-        Err(e) => {
-            error!("Failed to resolve address: {}", e);
-            return;
-        }
-    };
+    let warp_service = warp::service(routes);
 
-    warp::serve(routes).run(addr).await;
+    warp_lambda::run(warp_service)
```

Comment:
Ahhh I see, this is the only place where we use `warp_lambda`.

I think things are making more sense to me now. I was looking through the official [aws-lambda-rust-runtime](https://github.com/awslabs/aws-lambda-rust-runtime/tree/main) crate to see how things are typically done. There they have some examples, where they build the HTTP router using the framework's mechanism and just call [`lambda_http::run(router)`](https://docs.rs/lambda_http/latest/lambda_http/fn.run.html) in `main`. That works for them since the frameworks they use in their [examples](https://github.com/awslabs/aws-lambda-rust-runtime/blob/c4594f797c9c3a133845298eaa0a2f896ebe4220/examples/http-axum/src/main.rs) (typically with [`axum`](https://github.com/tokio-rs/axum)) are built on `tower`, and the router type in the framework implements the `tower` service traits. I believe `warp::service(routes)` returns a tower service, which means I think we can use `lambda_http::run(warp_service)` here as well (or something close to that). I'd prefer using `lambda_http` over `warp_lambda`, since the former is an official AWS supported crate. Moreover, `warp_lambda` just uses `lambda_http` [under the hood](https://docs.rs/warp_lambda/latest/src/warp_lambda/lib.rs.html#35-41), so it seems unnecessary.

Reply:
Okay, I'll make a ticket to remove it but I'm eager to just get this whole thing working. 

I looked at the `warp_lambda` crate and was not super impressed, and then realized it didn't matter if it worked which is where I am right now.

Reply:
Fair enough. Also, I might be mistaken about how easy it is to remove `warp_lambda`. As in I might be wrong about `warp::service` returning a tower service. I think it might actually return an hyper service or something.

Reply:
Hang on, I see you're saying it's possibly as easy as a swap;  let me try it.

Reply:
Okay, its more complicated than just a swap so I'm going to move it into a different ticket with a TODO since I don't want to spend time on that before the whole system works.

Reply:
Made https://github.com/stacks-network/sbtc/issues/276